x/staking/types/msg.go

```diff
@@ -295,7 +295,7 @@ func (msg MsgBeginRedelegate) ValidateBasic() sdk.Error {
 	if msg.ValidatorDstAddress.Empty() {
 		return ErrNilValidatorAddr(DefaultCodespace)
 	}
-	if msg.SharesAmount.LTE(sdk.ZeroDec()) {
+	if msg.Amount.Amount.LTE(sdk.ZeroInt()) {
```

Comment:
`.Amount.Amount` does not look nice. Maybe create a func `msg.RawAmount()`?

Reply:
Agree @melekes, sounds repetitive - though I believe `RawAmount` isn't much better than `Amount`.
I seeme to remember that we had faced this conundrum already before - didn't we agree on something different like `Value`? I may well be mistaken here @cwgoes

Reply:
Ughhh but `Value` is so generic. You're bonding/unbonding/redelgating an `amount`. I agree it's not pretty.

Reply:
Yeah this is a stumper, I think we need to just settle for `Amount.Amount` - using `Value` seems silly